### PR TITLE
Remove library config and admin server from Manager

### PR DESCRIPTION
**Pull request by andrewemeryanz**
This change does the following:

1. Removes the library config from the Manager interface in favour
of passing it in on server startup.

2. Removes the admin server from the Manager interface in favour of
including it in the library config.

3. Renames HandlerInitialiser to RestHandlerInitialiser

### CODE REVIEW

**Comment by andrewemeryanz** [comment]
Changes look good. Can you do the following:
- Add an end-to-end test of the admin server functionality similar to `TestValidRequestResponse`.
- Raise a separate pull request (I cannot comment nor approve my own request) then close this one.

File: core/callback.go [hunk 1]
```diff
@@ -30,4 +30,7 @@ type RestGenCallback interface {
 // GrpcGenCallback is currently a subset of RestGenCallback so is defined separately for convenience
 type GrpcGenCallback interface {
```
Comment (andrewemeryanz) [inline core/callback.go:36]: Also, please create a `GenCallback` that `RestGenCallback` and `GrpcGenCallback` both implement.

Comment (springwiz) [reply core/callback.go:36]: ok

**Comment by springwiz** [comment]
> Changes look good. Can you do the following:
> 
> * Add an end-to-end test of the admin server functionality similar to `TestValidRequestResponse`.
> * Raise a separate pull request (I cannot comment nor approve my own request) then close this one.

Didn't quite understand the first point, with AdminServer you mean the config that has been moved to LibraryConfig? but that's just config. What do you want me to test there?

**Comment by andrewemeryanz** [comment]
> Didn't quite understand the first point, with AdminServer you mean the config that has been moved to LibraryConfig? but that's just config. What do you want me to test there?

At the moment there's no test case that covers whether or not the admin server actually serves anything. The `TestValidRequestResponse` test case is an example of a test that starts up a server and validates that something is received.

**Review by springwiz — APPROVED** [review]
LGTM

**Pull request merged**